Packages/Netherlands3D/Windmills/Runtime/Scripts/Windmill.cs

```diff
@@ -0,0 +1,89 @@
+using System.Linq;
+using Netherlands3D.Events;
+using Netherlands3D.T3DPipeline;
+using UnityEngine;
+
+namespace Netherlands3D.Windmills
+{
+    public class Windmill : MonoBehaviour
+    {
+        private const string AXIS_HEIGHT_KEY = "asHoogte";
+        private const string ROTOR_DIAMETER_KEY = "rotorDiameter";
+        private const string STATUS_KEY = "status";
+
+        public float RotorDiameter { get; private set; }
+        public float AxisHeight { get; private set; }
+        public WindmillStatus Status { get; private set; }
+
+
+        [SerializeField] private GameObjectEvent onCityObjectVisualized;
+
+        [SerializeField] private GameObject windmillBase;
+        [SerializeField] private GameObject windmillAxis;
+        private Vector3 axisBasePosition;
+        [SerializeField] private GameObject windmillRotorConnection;
+        [SerializeField] private GameObject windmillRotor;
+        [SerializeField] private float rotationSpeed = 10f;
+
+        float fallbackHeight = 120f;
+        float fallbackDiameter = 120f;
+
+        private void Awake()
+        {
+            axisBasePosition = windmillAxis.transform.localPosition;
+        }
+
+        private void OnEnable()
+        {
+            onCityObjectVisualized.AddListenerStarted(Initialize);
+        }
+
+        public void Initialize(GameObject obj)
+        {
+            if (obj != transform.parent.gameObject)
+                return;
+
+            var cityObject = obj.GetComponent<CityObject>();
```

Comment:
Dit heeft nu een afhankelijkheid op CityObject. 
Kan dit script ook niet met public methods werken om de configuratie van de visual van de windmolen aan te passen zodat deze ook zonder CityObjects gebruikt kan worden?
Als we dit dan al als losse package aan gaan bieden lijkt het me mooier dat deze ook in te zetten zou kunnen zijn onafhankelijk van CityJSON/CityObjects.
Dat zou wel betekenen dat de package die custom objecten plaatst adhv. CityJSON op een andere manier (via UnityEvents? ) die data uitstuurt.

Reply:
Ik heb met Tom nog even gesproken of we de CityJSON afhandeling sowieso niet los moeten trekken van T3D en hier een losse package van zouden willen maken. Ik kan wel even kijken of ik door dependency inversion dat hier in deze package kan oplossen.